sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -137,13 +139,79 @@ public AnalyzedRelation analyze(Node node, Analysis analysis) {
+
+            // Use child relation to process expressions of the "root" Query node
+            statementContext.startRelation();
+            RelationAnalysisContext relationAnalysisContext = statementContext.currentRelationContext();
+            String qualifiedName = childRelation.getQualifiedName().toString();
+            // In case of Union we need to analyze ORDER BY based on the output symbols of the left-most relation
+            if (childRelation instanceof UnionSelect) {
+                qualifiedName = ((UnionSelect) childRelation).left().getQualifiedName().toString();
+            }
+            relationAnalysisContext.addSourceRelation(qualifiedName, childRelation);
+            ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+                functions,
+                statementContext.transactionContext(),
+                statementContext.convertParamFunction(),
+                new FullQualifiedNameFieldProvider(
+                    relationAnalysisContext.sources(),
+                    relationAnalysisContext.parentSources(),
+                    statementContext.transactionContext().sessionContext().defaultSchema()),
+                new SubqueryAnalyzer(this, statementContext));
+            ExpressionAnalysisContext expressionAnalysisContext = relationAnalysisContext.expressionAnalysisContext();
+            SelectAnalysis selectAnalysis = new SelectAnalysis(
+                childRelation.fields().size(),
+                relationAnalysisContext.sources(),
+                expressionAnalyzer,
+                expressionAnalysisContext);
+            for (Field field : childRelation.fields()) {
+                selectAnalysis.add(field.path(), field);
+            }
+            statementContext.endRelation();
+
+            return new OrderedLimitedRelation(
+                    childRelation,
+                    analyzeOrderBy(
+                        selectAnalysis,
+                        node.getOrderBy(),
+                        expressionAnalyzer,
+                        expressionAnalysisContext,
+                        false,
+                        false),
+                    longSymbolOrNull(node.getLimit(), expressionAnalyzer, expressionAnalysisContext),
+                    longSymbolOrNull(node.getOffset(), expressionAnalyzer, expressionAnalysisContext));
+        }
+        return process(node.getQueryBody(), statementContext);
     }
 
     @Override
     protected AnalyzedRelation visitUnion(Union node, StatementAnalysisContext context) {
-        throw new UnsupportedFeatureException("UNION is not supported");
+        if (node.isDistinct()) {
+            throw new UnsupportedFeatureException("UNION [DISTINCT] is not supported");
+        }
+        QueriedRelation first = (QueriedRelation) process(node.getLeft(), context);
+        QueriedRelation second = (QueriedRelation) process(node.getRight(), context);
+
+        // Check number of outputs
```

Comment:
The comment is kinda unnecessary as it's obvious from the code

Reply:
Btw, would be nice to use the `outputs()` method on the relation instead of the querySpec